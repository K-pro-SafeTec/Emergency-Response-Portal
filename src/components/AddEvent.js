import React, { Component } from 'react';
import { ParticipantsSelect } from './ParticipantsSelect';
import { TypeSelect } from './TypeSelect';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';


export class AddEvent extends Component {
  
  render() {
    const date = this.props.date;
    return (
      <div className="side-display">
        <h1>Legg til ny hendelse</h1>
        <form>
          <TextField
            id="date"
            label="Dato"
            type="date"
            defaultValue={getDateFormat(date)}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              step: 300, // 5 min
            }}
          />
        </form>
        <form>
          <TextField
            id="start"
            label="Start"
            type="time"
            defaultValue="09:00"
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              step: 300, // 5 min
            }}
          />
        </form>
        <form>
          <TextField
            id="end"
            label="Slutt"
            type="time"
            defaultValue="10:00"
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              step: 300, // 5 min
            }}
          />
        </form>
        <ParticipantsSelect/>
        <TypeSelect/>
        <div onClick={() => this.props.onSaveButtonClick()}>
          <Button variant="contained" color="primary">
            Lagre
          </Button>
        </div>
      </div>
    )
  }
}

function getDateFormat(date) {
  const year = date.getFullYear();
  let month = date.getMonth() + 1;
  if (month < 10) {
    month = "0" + month;
  }
  let dayInMonth = date.getDate();
  if (dayInMonth < 10) {
    dayInMonth = "0" + dayInMonth;
  }
  return year + "-" + month + "-" + dayInMonth
}