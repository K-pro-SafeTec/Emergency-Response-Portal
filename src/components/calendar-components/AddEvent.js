import React from 'react';
import PropTypes from 'prop-types';
import {ParticipantsSelect} from './ParticipantsSelect';
import {TypeSelect} from './TypeSelect';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import {getDateFormatYMD} from './../../helpers/calendar-helper';


export const AddEvent = (props) => {
  const date = props.date;
  const handleSubmit= () => {
    props.onSaveButtonClick(document.getElementById("date").value,
      document.getElementById("start").value,
      document.getElementById("end").value,
      document.getElementById("participants").value,
      document.getElementById("type").value)
  };
  return (
    <div className="side-display">
      <h1>Legg til ny hendelse</h1>
      <form onSubmit={handleSubmit}>
        <TextField id="date" label="Dato" type="date" defaultValue={getDateFormatYMD(date)}
          InputLabelProps={{
            shrink: true,
          }}
        />
        <TextField id="start" label="Start" type="time" defaultValue="09:00"
          InputLabelProps={{
            shrink: true,
          }}
          inputProps={{
            step: 300, // 5 min
          }}
        />
        <TextField id="end" label="Slutt" type="time" defaultValue="10:00"
          InputLabelProps={{
            shrink: true,
          }}
          inputProps={{
            step: 300, // 5 min
          }}
        />
        <ParticipantsSelect id="participants"/>
        <TypeSelect id="type"/>
      <div>
        <Button variant="contained" color="primary" type="submit">
          Lagre
        </Button>
      </div>
      </form>
    </div>
  )
};

AddEvent.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
  onSaveButtonClick: PropTypes.func.isRequired
};