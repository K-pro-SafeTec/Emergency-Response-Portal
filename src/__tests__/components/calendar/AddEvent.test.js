import React from 'react';
import AddEvent from "../../../components/calendar/AddEvent";
import TestRenderer from "react-test-renderer";
import {MemoryRouter as Router} from "react-router-dom";

let testRenderer;

beforeAll(() => {
  const mockOnClick = jest.fn();
  testRenderer = TestRenderer.create(<Router><AddEvent date={new Date(2018, 11)} onSaveButtonClick={mockOnClick} /></Router>);
});

test('renders correctly', () => {
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

test('contains two buttons', () => {
  const testInstance = testRenderer.root;
  const buttons = testInstance.findAllByType('button');
  expect(buttons.length).toBe(2);
});
