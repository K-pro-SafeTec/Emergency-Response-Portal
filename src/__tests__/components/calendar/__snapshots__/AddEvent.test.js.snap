// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="MuiTypography-root-1 MuiTypography-headline-6"
  >
    Legg til ny
     hendelse
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-36"
    >
      <label
        className="MuiFormLabel-root-47 MuiFormLabel-filled-51 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44 MuiInputLabel-shrink-43"
        data-shrink={true}
      >
        Dato
      </label>
      <div
        className="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-77 MuiInput-input-62 MuiInputBase-inputType-80 MuiInput-inputType-65"
          disabled={false}
          name="date"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
          value="2018-12-01"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-36"
    >
      <label
        className="MuiFormLabel-root-47 MuiFormLabel-filled-51 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44 MuiInputLabel-shrink-43"
        data-shrink={true}
      >
        Start
      </label>
      <div
        className="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-77 MuiInput-input-62 MuiInputBase-inputType-80 MuiInput-inputType-65"
          disabled={false}
          name="start"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          step={300}
          type="time"
          value="09:00"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-36"
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44 MuiInputLabel-shrink-43"
        data-shrink={true}
      >
        Slutt
      </label>
      <div
        className="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-77 MuiInput-input-62 MuiInputBase-inputType-80 MuiInput-inputType-65"
          defaultValue="10:00"
          disabled={false}
          name="end"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          step={300}
          type="time"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-36"
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
        data-shrink={false}
        htmlFor="age-simple"
      >
        Participants
      </label>
      <div
        className="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-84"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-85 MuiSelect-selectMenu-88 MuiInputBase-input-77 MuiInput-input-62"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            name="participants"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-91 MuiSelect-icon-90"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-36"
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
        data-shrink={false}
        htmlFor="age-simple"
      >
        Type
      </label>
      <div
        className="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-84"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-85 MuiSelect-selectMenu-88 MuiInputBase-input-77 MuiInput-input-62"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            name="type"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-91 MuiSelect-icon-90"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="mt32 flex spacebetween"
    >
      <button
        className="MuiButtonBase-root-130 MuiButton-root-104 MuiButton-contained-115 MuiButton-containedPrimary-116 MuiButton-raised-118 MuiButton-raisedPrimary-119"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-105"
        >
          Lagre
        </span>
        <span
          className="MuiTouchRipple-root-133"
        />
      </button>
      <button
        className="MuiButtonBase-root-130 MuiButton-root-104 MuiButton-text-106 MuiButton-textPrimary-107 MuiButton-flat-109 MuiButton-flatPrimary-110"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-105"
        >
          Avbryt
        </span>
        <span
          className="MuiTouchRipple-root-133"
        />
      </button>
    </div>
  </form>
</div>
`;
